src/Shared/Guard.cs

```diff
@@ -178,6 +192,84 @@ public static T ThrowIfNotOfType<T>([NotNull] object? value, [CallerArgumentExpr
             return result;
         }
 
+        /// <summary>
+        /// Throws an exception if the given view name is invalid according to the specification.
+        /// Null is valid because the instrument name will be used as the view name.
+        /// </summary>
+        /// <remarks>See specification: <see href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument"/>.</remarks>
+        /// <param name="viewName">The view name.</param>
+        /// <param name="paramName">The parameter name to use in the thrown exception.</param>
+        [DebuggerHidden]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ThrowIfInvalidViewName(string? viewName, [CallerArgumentExpression(nameof(viewName))] string? paramName = null)
+        {
+            if (!IsValidViewName(viewName))
+            {
+                throw new ArgumentException($"View name {viewName} is invalid.", paramName);
+            }
+        }
+
+        /// <summary>
+        /// Throws an exception if the given custom view name is invalid according to the specification.
+        /// </summary>
+        /// <remarks>See specification: <see href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument"/>.</remarks>
+        /// <param name="viewName">The view name.</param>
+        /// <param name="paramName">The parameter name to use in the thrown exception.</param>
+        [DebuggerHidden]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void ThrowIfInvalidCustomViewName([NotNull] string? viewName, [CallerArgumentExpression(nameof(viewName))] string? paramName = null)
+#pragma warning disable CS8777 // Parameter must have a non-null value when exiting.
+        {
+            if (!IsValidInstrumentName(viewName))
+            {
+                throw new ArgumentException($"Custom view name {viewName} is invalid.", paramName);
+            }
+        }
+#pragma warning restore CS8777 // Parameter must have a non-null value when exiting.
+
+        /// <summary>
+        /// Returns whether the given instrument name is valid according to the specification.
+        /// </summary>
+        /// <remarks>See specification: <see href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument"/>.</remarks>
+        /// <param name="instrumentName">The instrument name.</param>
+        /// <returns>Boolean indicating if the instrument is valid.</returns>
+        [DebuggerHidden]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool IsValidInstrumentName(string? instrumentName)
+        {
+            if (string.IsNullOrWhiteSpace(instrumentName))
+            {
+                return false;
+            }
+#if NET7_0_OR_GREATER
```

Comment:
There is no need to specify .NET version here. All supported .NT versions supports also generated regexes.

```suggestion
#if NET
            return InstrumentNameRegex().IsMatch(customViewName);
#else
            return InstrumentNameRegex.IsMatch(customViewName);
#endif
```

Reply:
Thanks, not needed anymore as per your suggestion.